### PR TITLE
succeed another mappings trawl

### PR DESCRIPTION
**Pull request by EnnuiL**
unfortunately, i couldn't manage to stab some class names even though the rest were mapped; we need backup

### CODE REVIEW

File: mappings/net/minecraft/block/Blocks.mapping [hunk 1]
```diff
@@ -29,9 +29,17 @@ CLASS net/minecraft/unmapped/C_qceyzrkd net/minecraft/block/Blocks
 		ARG 0 litLevel
 	METHOD m_idzyfosb (Lnet/minecraft/unmapped/C_txtbiemp;)I
 		ARG 0 state
+	METHOD m_ihepvmyd createPillarColor (Lnet/minecraft/unmapped/C_mlrprqox;Lnet/minecraft/unmapped/C_mlrprqox;Lnet/minecraft/unmapped/C_aevintex;)Lnet/minecraft/unmapped/C_djwplmtn;
```
Comment (LambdAurora) [inline mappings/net/minecraft/block/Blocks.mapping]: maybe `createPillarMapColor`? Just color is a bit ambiguous.

Comment (EnnuiL) [reply mappings/net/minecraft/block/Blocks.mapping]: *facepalms*
i might have wanted a `createPillarBlock` here, whoops

**Review by supersaiyansubtlety — APPROVED** [review]
Looks good except for the `createPillarColor` LambdaAurora already commented on

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 2]
```diff
@@ -576,6 +578,7 @@ CLASS net/minecraft/unmapped/C_triydqro net/minecraft/block/AbstractBlock
 		FIELD f_gepbjyhg nonSolid Z
 		FIELD f_gljcimhm emissiveLightingPredicate Lnet/minecraft/unmapped/C_triydqro$C_izsedryc;
 		FIELD f_hgzeyqby randomTicks Z
+		FIELD f_hzpexiar mapColorFunction Ljava/util/function/Function;
```
Comment (eristhea) [inline mappings/net/minecraft/block/AbstractBlock.mapping]: hmmm, maybe `mapColorGetter` to be a little more clear about what this does?

File: mappings/net/minecraft/structure/TrainRuinsGenerator.mapping [hunk 3]
```diff
@@ -0,0 +1,3 @@
+CLASS net/minecraft/unmapped/C_trcxslji net/minecraft/structure/TrainRuinsGenerator
```
Comment (eristhea) [inline mappings/net/minecraft/structure/TrainRuinsGenerator.mapping]: TRAIN RUINS GENERATOR


BOTTOM TEXT

**Review by eristhea — CHANGES_REQUESTED** [review]
I'd also appreciate naming the unmapped classes you've thrown in, but it's ok if not!

File: mappings/net/minecraft/structure/TrainRuinsGenerator.mapping (hunk 3 above)
Comment (EnnuiL) [reply mappings/net/minecraft/structure/TrainRuinsGenerator.mapping]: lmfao

**Comment by EnnuiL** [comment]
> I'd also appreciate naming the unmapped classes you've thrown in, but it's ok if not!

unfortunately, i'll have to leave the task to someone else; i'm not confident enough on mapping the class names

File: mappings/net/minecraft/block/AbstractBlock.mapping [hunk 4]
```diff
@@ -680,6 +692,10 @@ CLASS net/minecraft/unmapped/C_triydqro net/minecraft/block/AbstractBlock
 			ARG 2 pos
 		METHOD m_tzpskyoo (Lnet/minecraft/unmapped/C_txtbiemp;Lnet/minecraft/unmapped/C_peaveboq;Lnet/minecraft/unmapped/C_hynzadkk;)Z
 			ARG 0 state
+			ARG 1 world
+			ARG 2 pos
+		METHOD m_uggztwlf mapColor (Ljava/util/function/Function;)Lnet/minecraft/unmapped/C_triydqro$C_xnkxsdfy;
+			ARG 1 mapColorFunction
```
Comment (eristhea) [inline mappings/net/minecraft/block/AbstractBlock.mapping]:
```suggestion
			ARG 1 mapColorGetter
```

**Review by eristhea — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/block/AbstractBlock.mapping (hunk 4 above)
Comment (eristhea) [reply mappings/net/minecraft/block/AbstractBlock.mapping]: NOOOOOOOOO two seconds too late

**Review by eristhea — APPROVED** [review]

File: mappings/net/minecraft/block/AbstractBlock.mapping (hunk 4 above)
Comment (EnnuiL) [reply mappings/net/minecraft/block/AbstractBlock.mapping]: *evil laugh*

**Pull request merged**